pkg/transformer/kubernetes/k8sutils.go

```diff
@@ -441,11 +441,11 @@ func (k *Kubernetes) UpdateKubernetesObjects(name string, service kobject.Servic
 
 		// Configure the container restart policy.
 		switch service.Restart {
-		case "", "always":
+		case "", "always", "any":
 			template.Spec.RestartPolicy = api.RestartPolicyAlways
-		case "no":
+		case "no", "none":
 			template.Spec.RestartPolicy = api.RestartPolicyNever
-		case "on-failure":
+		case "on-failure", "on_failure":
```

Comment:
so found out it was error upstream. it's fixed now :( so we can remove `on_failure`. Sorry for confusion!

https://github.com/docker/cli/pull/262